projects/step-frontend/src/lib/modules/execution/components/execution-page/execution-page.component.ts

```diff
@@ -0,0 +1,143 @@
+import { Component, OnDestroy, OnInit } from '@angular/core';
+import { downgradeComponent, getAngularJSGlobal } from '@angular/upgrade/static';
+import { AJS_MODULE, Execution } from '@exense/step-core';
+import { ExecutionTab } from '../../shared/execution-tab';
+
+@Component({
+  selector: 'step-execution-page',
+  templateUrl: './execution-page.component.html',
+  styleUrls: ['./execution-page.component.scss'],
+})
+export class ExecutionPageComponent implements OnInit, OnDestroy {
+  listTab: ExecutionTab = { label: 'Executions', type: 'list', id: 'list', title: 'Executions List' };
+  tabs: ExecutionTab[] = [this.listTab];
+  activeTab!: ExecutionTab;
+
+  private locationChangeFunction = () => {
+    let urlParts = this.getUrlParts();
+    const executionId = urlParts[1];
+    if (this.activeTab.id === executionId) {
+      // the tab didn't change, but just the subcontent
+      return;
+    }
+    if (executionId) {
+      let existingTab = this.tabs.find((tab) => tab.id === executionId);
+      if (existingTab) {
+        this.switchTab(existingTab);
+      } else {
+        let executionTab: ExecutionTab = {
+          id: executionId,
+          active: true,
+          label: executionId,
+          type: 'progress',
+          title: executionId,
+        };
+        this.tabs.push(executionTab);
+        this.switchTab(executionTab);
+      }
+    } else {
+      this.switchTab(this.listTab);
+    }
+  };
+
+  updateUrl() {
+    let url = this.createUrl(this.activeTab);
+    // window.history.replaceState({}, '', url);
+    location.href = url;
+  }
+
+  createUrl(tab: ExecutionTab): string {
+    const url = window.location.href;
+    let index = url.indexOf('executions');
+    let urlItems = url.substring(index).split('/');
+    urlItems[1] = tab.id;
+    if (tab.subTab) {
+      urlItems[2] = tab.subTab;
+    } else {
+      urlItems.length = 2;
+    }
+    return url.substring(0, index) + urlItems.join('/');
+  }
+
+  handleTabChange(tabId: string) {
+    let newTab = this.tabs.find((tab) => tab.id === tabId);
+    if (newTab) {
+      this.switchTab(newTab);
+    }
+  }
+
+  handleSubTabChange(tabId: string, subTab: string): void {
+    let tab = this.tabs.find((tab) => tab.id === tabId);
+    if (tab) {
+      tab.subTab = subTab;
+    }
+    this.replaceUrlSubTab(subTab);
+  }
+
+  handleTabTitleUpdate(event: { eId: string; execution: Execution }) {
+    let tab = this.tabs.find((tab) => tab.id === event.eId);
+    if (tab) {
+      tab.title = event.execution.description;
+    }
+  }
```

Comment:
In comparison with original implantation:
```
    $scope.updateTabTitle = function (eid, execution) {
      var tab = _.findWhere($scope.tabs, { id: eid });
      tab.title = execution.description;
      tab.execution = execution;
    };
```
Execution reference inside the tab object is required to determine it's status and display in the tab (running, success, error, etc.).
Currently now status isn't display in the tab bar.
Please add the execution assignment to fix it.